src/test/scala/uk/gov/hmrc/ui/specs/LeaseOutYourPetsSpec.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import uk.gov.hmrc.ui.specs.BaseSpec
+
+class LeaseOutYourPetsSpec extends BaseSpec {
+  System.setProperty("browser", "chrome") // temp, just to make it easier to run from IDE
+
+  info("As a pet owner")
+  info("I want to be able to lease out my pet")
+  info("So that I can supplement my income")
+
+  Feature("Sign-up your pet") {
+    Scenario("Sign-up is successful, when all information can be provided") {
+      // I start the sign-up process
+      // I am asked to enter my pets name
+      // I enter my pets name
+      // I am asked if my pet is vaccinated
+      // I select yes
+      // I am asked when my pets next vaccination is due
```

Comment:
how would you write this in scalatest when using featurespec? Like the following?
```suggestion
      // When: I start the sign-up process
      // Then: I am asked to enter my pets name
      // When: I enter my pets name
      // Then: I am asked if my pet is vaccinated
      // When: I select yes
      // Then: I am asked when my pets next vaccination is due
```

Reply:
And how would you want to see the steps encoded, like this?
```scala
When("I start the sign-up process")
SignUpPetJourney.start()

Then("I am asked to enter my pets name")
on WhatIsYourPetsNamePage

When("I enter my pets name")
WhatIsYourPetsNamePage.enterName("bob")

Then("I am asked if my pet is vaccinated")
on IsYourPetVaccinatedPage

When("I select yes")
IsYourPetVaccinatedPage.selectYes()
```

would you want to see the on or have it more like this, where I guess you'd need to check you're on the right page when interacting with the page or throw an exception that you're not on the expected page.

```scala
When("I start the sign-up process")
SignUpPetJourney.start()

Then("I am asked to enter my pets name")
When("I enter my pets name")
WhatIsYourPetsNamePage.enterName("bob")

Then("I am asked if my pet is vaccinated")
When("I select yes")
IsYourPetVaccinatedPage.selectYes()
```